nuclio_sdk/platform.py

```diff
@@ -51,11 +51,13 @@ async def explicit_ack(self, qualified_offset):
                 "Cannot send explicit ack since control callback was not initialized"
             )
 
-    def on_signal(self, callback, sig=signal.SIGTERM):
+    def set_termination_callback(self, callback):
         """
-        Syntactic sugar to bind an incoming system signal on user's callback
+        Register a callback to be called when the platform is terminating
```

Comment:
I just realized that this language actually implies that one may register multiple callbacks, which is not the case. We may also want to mention that the callback is nullary (will be called with zero arguments).